### PR TITLE
Move most static metadata to pyproject.toml.

### PR DESCRIPTION
**Pull request by Sachaa-Thanasius**
## Summary

It moves most of the library's metadata to pyproject.toml from setup.py, excluding the library version due to its particular type of dynamism and the fact that code would seemingly be required for it regardless of possible alternatives like setuptools_scm.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Sachaa-Thanasius** [comment]
Hmm. I'm not sure what's causing the issue here, but I'd guess that moving the data somehow . . . changed the impact of manually setting `PYTHONPATH` to the path of the working directory? Running the same tests on my machine works just fine after an editable install, without the `PYTHONPATH="$(pwd)"` portion of the command.

**Comment by Sachaa-Thanasius** [comment]
@Rapptz Any pointers on what's causing these tests to fail? I'm guessing it has to do with the explicit `PYTHONPATH` since `aiohttp` can't be found for importing, and the tests sans `PYTHONPATH` seem to work fine locally.

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,84 @@
 [build-system]
-requires = ["setuptools", "wheel"]
+requires = ["setuptools"]
```
Comment (Sachaa-Thanasius) [inline pyproject.toml:2]: setuptools automatically installs wheel if needed. Further info is in the first note in this section: https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use

**Review by Rapptz — APPROVED** [review]
Thanks. Let's give this a try.

**Pull request merged**